src/main/java/hudson/model/ExternalJob.java

```diff
@@ -76,7 +78,11 @@ public synchronized ExternalRun newBuild() throws IOException {
         long timeSinceLast = System.currentTimeMillis() - lastBuildStartTime;
         if (timeSinceLast < 1000) {
             try {
-                Thread.sleep(1000 - timeSinceLast);
+                while(timeSinceLast > 0){
+                    wait();
```

Comment:
I was "hudson.model.ExternalJob.newBuild() calls Thread.sleep() with a lock held" which I thought was sort of the point.  I suppressed the findbugs and converted back; I didn't really like this solution.